dao/src/main/java/com/epam/esm/dao/CertificateDao.java

```diff
@@ -0,0 +1,63 @@
+package com.epam.esm.dao;
+
+import com.epam.esm.dao.entity.Certificate;
+import com.epam.esm.dao.exception.NoSuchResultException;
+import com.epam.esm.dao.impl.util.CertificateSearchCriteria;
+
+import java.util.List;
+
+/** The interface Certificate dao. */
+public interface CertificateDao {
```

Comment:
As I see we have dao classes with the same structure of methods -> could we create general interface for all dao interfaces?